src/lib.rs

```diff
@@ -698,14 +698,19 @@ impl AstarteSdk {
             let data: AstarteType = data.into();
 
             let interfaces = self.interfaces.read().await;
+            let interface_major = interfaces
+                .interfaces
+                .get(interface_name)
+                .ok_or_else(|| AstarteError::SendError("Interface doesn't exist".into()))?
```

Comment:
Isn't the existence of the interface verified earlier? E.g. [here](https://github.com/astarte-platform/astarte-device-sdk-rust/blob/5d4ec0a5dc5736382e2f27d53fb1be1f2892f6f8/src/lib.rs#L632-L634)

Reply:
It's a safer alternative than `.unwrap()`, but I agree it's redundant.

Reply:
Since it is used in a separate function, perhaps it is better to keep a safer version.